### PR TITLE
raptor: remove weakly dominated solutions and (now useless) b_dest

### PR DESCRIPTION
**Pull request by TeXitoi**
Solutions that end at the same time, but begin later and with more transfer will not be found with this version. Performances are better than before (but this performance improvement must not be assumed acquired). See the commented test for an example.

### CODE REVIEW

File: source/routing/raptor_utils.h [hunk 1]
```diff
@@ -135,74 +135,5 @@ struct Labels {
...
-    }
-
-    void reinit(const std::vector<type::StopPoint*>& stops, const DateTime& bound) {
-        reinit(stops);
-        best_now = bound;
-    }
-
-};
-
 } // namespace routing
```
Comment (l-vincent-l) [inline source/routing/raptor_utils.h:99]: Pourquoi tu ne supprimes pas tout le fichier ?

Comment (TeXitoi) [inline source/routing/raptor_utils.h:99]: Parce que sinon, il n'y a plus de labels, et que sans labels, raptor, il marche pas top ;)

Comment (l-vincent-l) [inline source/routing/raptor_utils.h:99]: Je vois ce que tu veux dire.
Je m'étais embrouillé entre le .h et le .cpp

File: source/routing/raptor.cpp [hunk 2]
```diff
@@ -79,16 +80,15 @@ bool RAPTOR::apply_vj_extension(const Visitor& v, const bool global_pruning,
...
             auto sp = st.journey_pattern_point->stop_point;
             const auto sp_idx = SpIdx(*sp);
-            const DateTime bound = (v.comp(best_labels_pts[sp_idx], b_dest.best_now) || !global_pruning) ?
-                                        best_labels_pts[sp_idx] : b_dest.best_now;
-            if (v.comp(bound, workingDt)) { continue; }
-            if (!v.comp(workingDt, working_labels.dt_pt(sp_idx))) { continue; }
 
-            working_labels.mut_boarding_jpp_pt(sp_idx) = state.boarding_jpp_idx;
-            working_labels.mut_used_jpp(sp_idx) = JppIdx(*st.journey_pattern_point);
+            // because of performance reason, we desactivate weakly
```
Comment (antoine-de) [inline source/routing/raptor.cpp]: s/because of/for/ no ?
or s/reason/issues/ nop ? this seems strange

File: source/routing/raptor.cpp [hunk 3]
```diff
@@ -281,11 +263,10 @@ RAPTOR::compute_nm_all(const std::vector<std::pair<type::EntryPoint, vec_stop_po
         for (const auto& n_stop_point : n_point.second)
             calc_dep.push_back(n_stop_point);
 
-    auto calc_dep_solutions = init_departures(calc_dep, departure_datetime, clockwise, disruption_active);
     clear(clockwise, bound);
-    init(calc_dep_solutions, {}, bound, clockwise); // no exit condition (should be improved)
+    init(calc_dep, departure_datetime, clockwise, accessibilite_params.properties);
```
Comment (antoine-de) [inline source/routing/raptor.cpp:267]: don't you want to call first_raptor_loop there ?

File: source/routing/raptor.cpp [hunk 4]
```diff
@@ -341,7 +294,7 @@ RAPTOR::compute_nm_all(const std::vector<std::pair<type::EntryPoint, vec_stop_po
 }
 
 void
-RAPTOR::isochrone(const vec_stop_point_duration &departures_,
+RAPTOR::isochrone(const vec_stop_point_duration &departures,
```
Comment (antoine-de) [inline source/routing/raptor.cpp]: :poop:

File: source/routing/raptor.cpp (hunk 3 above)
Comment (TeXitoi) [inline source/routing/raptor.cpp:267]: I'm afraid of NM...

File: source/routing/raptor_api.cpp [hunk 5]
```diff
@@ -856,59 +856,26 @@ make_nm_response(RAPTOR &raptor, const std::vector<type::EntryPoint> &origins,
...
-                    if (path.origin.type != nt::Type_e::Unknown)
-                        n_entry_point = path.origin;
-                }
-                if (n_entry_point.type == nt::Type_e::Unknown)
-                    continue;
+                               accessibilite_params, forbidden, clockwise);
 
-                const type::EntryPoint& m_entry_point = paths_for_m_point.first;
+        // compute isochron style result at "m point"
+        for(std::pair<type::EntryPoint, std::vector<Path>> paths_for_m_point : paths_by_entrypoint) {
```
Comment (antoine-de) [inline source/routing/raptor_api.cpp]: you need to const& it no ?

File: source/routing/raptor_api.cpp [hunk 6]
```diff
@@ -856,59 +856,26 @@ make_nm_response(RAPTOR &raptor, const std::vector<type::EntryPoint> &origins,
...
-                    std::reverse(paths.begin(), paths.end());
-                    worker.init(n_entry_point, {m_entry_point});
-                    add_pathes(enhanced_response, paths, raptor.data, worker, n_entry_point, m_entry_point, datetimes, clockwise, show_codes);
-                }
-                else {
-                    worker.init(m_entry_point, {n_entry_point});
-                    add_pathes(enhanced_response, paths, raptor.data, worker, m_entry_point, n_entry_point, datetimes, clockwise, show_codes);
-                }
+            std::vector<type::StopPoint*> stop_points;
+            for(auto& path : paths) {
```
Comment (antoine-de) [inline source/routing/raptor_api.cpp]: const too

**Pull request merged**